git-cliff-core/src/changelog.rs

```diff
@@ -1044,6 +1079,7 @@ chore(deps): fix broken deps
 		changelog.generate(&mut out)?;
 		expect_test::expect![[r#"
     # Changelog
+
```

Comment:
What are the reason for these newline changes? Can we preserve the default behavior?

Reply:
This is caused by the difficult handling of newlines at the end of the header. I'll look into it.

Reply:
I could not find a straight forward way of preserving the new-line handling. It boils down to whether we use `write` or `writeln` when rendering the header template in [changelog.rs#513](https://github.com/Cyclonit/git-cliff/blob/feature/make_changelog_header_a_template/git-cliff-core/src/changelog.rs#L513).

- When using `write` (no additional `\n` after the header), the tests in `changelog.rs` are successful. They do not expect a new line between the header and body.

- When using `writeln` (forced additional `\n` after the header`), the test fixtures are successful. All fixtures that contain a header expect a new line between the header and body.

But in either case, the respective other tests fail. Personally I prefer the option of using `writeln`. In almost all use cases, the body begins with a header. Having a new line in front of it results in cleaner markdown. This would require changing the tests in `changelog.rs` but leaves the fixtures unchanged. It is also unlikely to have an impact on current users.

Reply:
Yeah, I'm on the same page. We can simply go with `println!` and update the tests in `changelog.rs`. Like you said, it is not likely to have a big impact.